nvflare/private/fed/app/deployer/simulator_deployer.py

```diff
@@ -69,44 +71,44 @@ def create_fl_client(self, client_name, args):
         deployer.build(build_ctx)
         federated_client = deployer.create_fed_client(args)
 
-        client_engine = SimulatorParentClientEngine()
-        federated_client.set_client_engine(client_engine)
-        federated_client.register()
-        federated_client.start_heartbeat()
-        federated_client.run_manager = None
+        self._create_client_cell(client_config, client_name, federated_client)
 
-        return federated_client, client_config, args
+        federated_client.register()
 
-    def create_admin_agent(self, server_args, federated_client: FederatedClient, args, rank=0):
-        sender = AdminMessageSender(
-            client_name=federated_client.token,
-            server_args=server_args,
-            secure=False,
-        )
-        client_engine = SimulatorClientEngine(federated_client, federated_client.token, sender, args, rank)
-        admin_agent = FedAdminAgent(
-            client_name="admin_agent",
-            sender=sender,
-            app_ctx=client_engine,
-        )
-        admin_agent.app_ctx.set_agent(admin_agent)
+        client_engine = SimulatorParentClientEngine(federated_client, federated_client.token, args)
         federated_client.set_client_engine(client_engine)
-        for processor in ClientRequestProcessors.request_processors:
-            admin_agent.register_processor(processor)
+        # federated_client.start_heartbeat()
+        federated_client.run_manager = None
 
-        client_engine.fire_event(EventType.SYSTEM_START, client_engine.new_context())
+        return federated_client, client_config, args, build_ctx
+
+    def _create_client_cell(self, client_config, client_name, federated_client):
+        target = client_config["servers"][0].get("service").get("target")
+        scheme = client_config["servers"][0].get("service").get("scheme", "grpc")
+        credentials = {}
+        parent_url = None
+        cell = Cell(
+            fqcn=client_name,
+            root_url=scheme + "://" + target,
+            secure=self.secure_train,
+            credentials=credentials,
+            create_internal_listener=True,
+            parent_url=parent_url,
+        )
+        cell.start()
+        federated_client.cell = cell
+        federated_client.communicator.cell = cell
+        # if self.engine:
+        #     self.engine.admin_agent.register_cell_cb()
```

Comment:
Instead of creating and set the cell for federated_client and its communicator.
I think it is better to create the cell when that federated_client object is created.

Reply:
The cell creation is after the client object is created. Not like the same way on the server. Need to set to client after the creation.